pinot-core/src/test/java/org/apache/pinot/core/data/recordtransformer/DataTypeTransformerTest.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.data.recordtransformer;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertEqualsNoOrder;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.fail;
+
+
+public class DataTypeTransformerTest {
+  private static final String COLUMN = "testColumn";
+
+  @Test
+  public void testStandardize() {
+    // Tests for Map
```

Comment:
Can we please add tests for the exact case where we were seeing the problem? The AvroRecorExtractor extracted the column as Object[] where each value in the array is a Map of single KV pair and this is what is passed to DataTypeTransformer.  The actual type of column in the schema is MV STRING and the transformer should just convert it into Object[] where each value is a String and is the corresponding value from each Map. 

Looking at the code in standardize and standardizeCollection, it looks the above scenario is well taken care of but want to make sure that in future when this code is changed for either handling map probably or something else, we don't regress again. 

@jackjlli , is there any other case that we were seeing when the problem was original reproduced. Good to share a sample of original data so that we can use it in tests here,

Reply:
@siddharthteotia Added

Reply:
I think Line 138 in `DataTypeTransformerTest.java` is the example that we expect.